gsad/src/gsad_gmp.c

```diff
@@ -423,8 +414,8 @@ filter_exists (gvm_connection_t *connection, const char *filt_id)
   entity_t entity;
 
   if (filt_id == NULL
-      || strcmp (filt_id, "0")
-      || strcmp (filt_id, "-2"))
+      || str_equal (filt_id, "0")
+      || str_equal (filt_id, "-2"))
```

Comment:
I think you forgot to negate the results here.
Also, you could use FILT_ID_NONE and FILT_ID_USER_SETTING here.

Reply:
Thanks for the hint! Didn't know we introduced those defines already. Nice! I'll check the condition.

Reply:
Regarding the condition. IMHO it was broken before and my change fixed it. Can you take another look?

Reply:
I've checked the context again and it looks like the condition was wrong.